moveit_kinematics/test/launch/fanuc-kdl-singular.test.py

```diff
@@ -0,0 +1,199 @@
+import launch_testing
+import os
+import pytest
+import unittest
+import yaml
+from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch_ros.actions import Node
+from launch_testing.util import KeepAliveProc
+
+
+def load_file(package_name, file_path):
+    package_path = get_package_share_directory(package_name)
+    absolute_file_path = os.path.join(package_path, file_path)
+
+    try:
+        with open(absolute_file_path, "r") as file:
+            return file.read()
+    except EnvironmentError:  # parent of IOError, OSError *and* WindowsError where available
+        return None
+
+
+def load_yaml(package_name, file_path):
+    package_path = get_package_share_directory(package_name)
+    absolute_file_path = os.path.join(package_path, file_path)
+
+    try:
+        with open(absolute_file_path, "r") as file:
+            return yaml.full_load(file)
+    except EnvironmentError:  # parent of IOError, OSError *and* WindowsError where available
+        return None
+
+
+@pytest.mark.rostest
+def generate_test_description():
+
+    robot_description_config = load_file(
+        "moveit_resources_fanuc_description", "urdf/fanuc.urdf"
+    )
+    robot_description = {"robot_description": robot_description_config}
+
+    robot_description_semantic_config = load_file(
+        "moveit_resources_fanuc_moveit_config", "config/fanuc.srdf"
+    )
+    robot_description_semantic = {
+        "robot_description_semantic": robot_description_semantic_config
+    }
+    kinematics_yaml = load_yaml(
+        "moveit_resources_fanuc_moveit_config", "config/kinematics.yaml"
+    )
+    robot_description_kinematics = {"robot_description_kinematics": kinematics_yaml}
+    test_param = load_yaml("moveit_kinematics", "config/fanuc-kdl-test.yaml")
+
+    singular_private_params = {
+        "seed": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+        "num_fk_tests": 0,
+        "num_ik_tests": 0,
+    }
+
+    singular_unit_tests_poses = {
```

Comment:
Could you move these poses into a yaml file, maybe even the new test yamls that you created?

Reply:
Sure